frontend/src/app/service/http-service/transformation-http.service.ts

```diff
@@ -40,6 +41,57 @@ export class TransformationHttpService {
     return this.httpClient.put<Transformation>(url, transformation);
   }
 
+  public updateExpandComponent(
+    transformation: Transformation
+  ): Observable<Transformation> {
+    const url = `${this.apiEndpoint}/transformations/${transformation.id}?update_component_code=true&expand_component_code=true`;
```

Comment:
QueryParams sollten als HttpParams-Objekt im HTTP-Request stehen und nicht als String in der URL.